testing/playwright/fixtures/helpers/resourceCreationHelpers.ts

```diff
@@ -0,0 +1,289 @@
+import type { Page } from '@playwright/test';
+
+import { CreatePlanWizardPage } from '../../page-objects/CreatePlanWizard/CreatePlanWizardPage';
+import { CreateProviderPage } from '../../page-objects/CreateProviderPage';
+import { PlanDetailsPage } from '../../page-objects/PlanDetailsPage/PlanDetailsPage';
+import { EndpointType, ProviderType } from '../../types/enums';
+import { createPlanTestData, type ProviderData } from '../../types/test-data';
+import { getProviderConfig } from '../../utils/providers';
+import { MTV_NAMESPACE } from '../../utils/resource-manager/constants';
+import type { ResourceManager } from '../../utils/resource-manager/ResourceManager';
+
+export type TestProvider = V1beta1Provider & {
+  metadata: {
+    name: string;
+    namespace: string;
+  };
+};
+
+export type TestPlan = V1beta1Plan & {
+  metadata: {
+    name: string;
+    namespace: string;
+  };
+  testData: ReturnType<typeof createPlanTestData>;
+};
+
+export const createProvider = async (
+  page: Page,
+  resourceManager: ResourceManager,
+  namePrefix = 'test-provider',
+  providerKey?: string,
+): Promise<TestProvider> => {
+  const uniqueId = crypto.randomUUID();
+  const providerName = `${namePrefix}-${uniqueId}`;
+
+  const key = providerKey ?? process.env.VSPHERE_PROVIDER ?? 'vsphere-8.0.1';
+  const providerConfig = getProviderConfig(key);
+
+  if (!providerConfig) {
+    throw new Error(`Provider configuration not found for key: ${providerKey}`);
+  }
+
+  const testProviderData: ProviderData = {
+    name: providerName,
+    projectName: MTV_NAMESPACE,
+    type: providerConfig.type,
+    endpointType: providerConfig.endpoint_type ?? EndpointType.VCENTER,
+    hostname: providerConfig.api_url,
+    username: providerConfig.username,
+    password: providerConfig.password,
+    vddkInitImage: providerConfig.vddk_init_image,
+  };
+
+  // For OVA providers, create directly via API with applianceManagement setting
+  if (providerConfig.type === ProviderType.OVA) {
+    // Navigate to console to establish session (needed for CSRF token)
+    const createProviderPage = new CreateProviderPage(page, resourceManager);
+    await createProviderPage.navigationHelper.navigateToConsole();
+
+    // Create minimal secret (CRD requires spec.secret, OVA only needs URL, no credentials)
+    const secretName = `${providerName}-secret`;
+    const secret: IoK8sApiCoreV1Secret = {
+      apiVersion: 'v1',
+      kind: 'Secret',
+      metadata: {
+        name: secretName,
+        namespace: MTV_NAMESPACE,
+      },
+      type: 'Opaque',
+      stringData: {
+        url: testProviderData.hostname,
+      },
+    };
+
+    const createdSecret = await resourceManager.createSecret(page, secret);
+    if (!createdSecret) {
+      throw new Error(`Failed to create secret for OVA provider ${providerName}`);
+    }
+
+    // Create provider with applianceManagement setting
+    const provider: V1beta1Provider = {
```

Comment:
here too can we encapsulate the provider creation? something like createProvider(metadataObj, specObj)?